### PR TITLE
WebTestCase performance optimization

### LINKED ISSUE #170: Use core patch to Improve WebTestCase performance [4h]
https://www.drupal.org/node/2759197#comment-11385805

The folks in backdrop are using it, and their entire core test suite runs in under 4 minutes (8 in total, including the setup)

### PR DESCRIPTION
**Pull request by AronNovak**
will fix #170 

## Status
merely the patch is there

## Todo
check with real tests, check if any further steps are needed

### CODE REVIEW

**Comment by amitaibu** [comment]
Note:

> The thing is opt-in. You will need to put --cache for the test runner for that to work.

**Comment by AronNovak** [comment]
**0.3/4h**

What's left:
 - add a basic Simpletest into the Hedley group
 - make sure Travis calls Simpletest correctly
 - check the benefit of the patch

**Comment by amitaibu** [comment]
> add a basic Simpletest into the Hedley group

You can run core's tests

File: ci-scripts/docker_files/run_simpletest.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+source helper_functions.sh
+
+# -------------------------------------------------- #
+# Run SimpleTest
+# -------------------------------------------------- #
+print_message "Run SimpleTest."
+export PATH="$HOME/.composer/vendor/bin:$PATH"
+drush @server.local en simpletest -y
+cd $ROOT_DIR/server
+php ./www/scripts/run-tests.sh --cache --php $(which php) --concurrency 4 --verbose --color --url http://server.local Hedley 2>&1 | tee /tmp/simpletest-result.txt
```
Comment (bitamar) [inline ci-scripts/docker_files/run_simpletest.sh]: According to this comment: https://www.drupal.org/node/2759197#comment-11970372 , adding `--cache-modules` should save a lot of time

Comment (AronNovak) [reply ci-scripts/docker_files/run_simpletest.sh]: Good point, i will apply it too afterwards everything is stable with one option.

**Comment by AronNovak** [comment]
Actually there are tests in the Hedley group:
image


That's likely a good opportunity to fix (or drop) these tests (in another PR).

**Comment by bitamar** [comment]
I think I added these tests. I'll have a look.

**Comment by AronNovak** [comment]
Local performance test on SimpleTest group:
image
without the `--cache` option.

image
with the `--cache` option.

image
with the `--cache` and `--cache-modules` option.

so the final execution time is 77% of the original one at localhost. Maybe in the cloud, with worse IO, the benefit is even greater, going to check soon.

**Comment by AronNovak** [comment]
image

So the same three cases are being tested right now in the cloud.

**Comment by bitamar** [comment]
The Hedley tests should be passing with #174

**Comment by AronNovak** [comment]
Test results from Travis with SimpleTest test case (Hedley is not meaningful with only one test):

## no caching
https://travis-ci.org/Gizra/drupal-elm-starter/jobs/283675496
```
 > Starting SimpleTest tests 
Thu Oct  5 13:01:16 UTC 2017
> Finished SimpleTest tests 
Thu Oct  5 13:03:22 UTC 2017
```

Total time: 126 seconds **(100%)**

## `--cache` option
https://travis-ci.org/Gizra/drupal-elm-starter/jobs/283674325
```
> Starting SimpleTest tests 
Thu Oct  5 12:48:20 UTC 2017

 > Finished SimpleTest tests 
Thu Oct  5 12:49:45 UTC 2017
```

Total time: 85 seconds **(67%)**

## `--cache` `--cache-modules` options
https://travis-ci.org/Gizra/drupal-elm-starter/jobs/283674971
```
> Starting SimpleTest tests 
Thu Oct  5 12:57:20 UTC 2017
 > Finished SimpleTest tests 
Thu Oct  5 12:58:40 UTC 2017
```

Total time: 61 seconds **(48%)**

That's impressive, I must say.

**Comment by AronNovak** [comment]
I think it"s ready for review, and then, ready to propagate to all projects with non-trivial SimpleTest coverage (where there's only one single test case, like at Hedley group here, it does not help actualy).

**Comment by amitaibu** [comment]
👍 MWG

**Pull request merged**